percy/client.py

```diff
@@ -13,7 +14,8 @@ class Client(object):
     def __init__(self, connection=None, config=None, environment=None):
         self._environment = environment if environment else Environment()
         self._config = config if config else Config()
-        self._connection = connection if connection else Connection(self.config)
+        self._user_agent = str(UserAgent(self))
```

Comment:
One more comment here — when I see the `user_agent` variable later on in the code, I think it's a UserAgent object, but it's actually not because you serialize it here and pass only the string down.

You could pass down the UserAgent object all the way through into Connection, and serialize it in there for a bit more clarity. This also addresses this failure mode: This `str()` call is actually super important because it's the thing that incurs the expensive `import` calls — because you're doing it here, if I were to construct a different `Connection` object and pass it in to Client, Connection would end up either erroring or running the imports every single HTTP request. To fix this, definitely should push the `str()` handling down into Connection, so that Connection handles the serialization of UserAgent no matter what.